provider/vsphere/provider.go

```diff
@@ -69,9 +71,39 @@ func (p environProvider) CloudSchema() *jsonschema.Schema {
 
 // Ping tests the connection to the cloud, to verify the endpoint is valid.
 func (p environProvider) Ping(endpoint string) error {
-	// a stub for now until we get a real implementation.
-	_, err := url.Parse(endpoint)
-	return errors.Trace(err)
+	// try to be smart and not punish people for adding or forgetting http
+	var u *url.URL
+	var err error
+	if strings.HasPrefix(strings.ToLower(endpoint), "http") {
+		u, err = url.Parse(endpoint)
+		if err != nil {
+			return errors.New("Invalid endpoint format, please give a full url or IP/hostname.")
+		}
+	} else {
+		u, err = url.Parse("https://" + endpoint + "/sdk")
+		if err != nil {
+			return errors.New("Invalid endpoint format, please give a full url or IP/hostname.")
+		}
+	}
+
+	client, logout, err := newConnection(u)
+	if err != nil {
+		return errors.Errorf("No VSphere server running at %s", endpoint)
+	}
+	defer logout()
+	err = client.Login(context.TODO(), nil)
+	if err == nil {
+		// shouldn't happen, but can't really complain if it does.
+		return nil
+	}
+	// There's no way to get at any type information in the returned error, so
+	// we have to just look at the string value.
+	if err.Error() == "ServerFaultCode: Cannot complete login due to an incorrect user name or password." {
```

Comment:
Make that string into a const somewhere, we cant ensure we will not need various strings as vsphere evolves and only one string in an if is coming back to bite us.